test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -145,10 +153,345 @@ public boolean canRunAsync() {
+								.treasury(beneficiary),
+						tokenCreate(otherFungibleToken)
+								.tokenType(FUNGIBLE_COMMON)
+								.treasury(beneficiary),
+						tokenCreate(uniqueToken)
+								.tokenType(NON_FUNGIBLE_UNIQUE)
+								.supplyKey(multiPurpose)
+								.initialSupply(0L)
+								.treasury(treasury),
+						mintToken(uniqueToken, List.of(copyFromUtf8("ONE"), copyFromUtf8("TWO"))),
+						getAccountInfo(beneficiary).savingSnapshot(beneficiary),
+						getAccountInfo(treasury).savingSnapshot(treasury)
+				).when(
+						cryptoTransfer(
+								movingUnique(uniqueToken, 1L)
+										.between(treasury, beneficiary),
+								moving(500, fungibleToken).between(beneficiary, treasury)
+						).via(transferTxn).hasKnownStatus(ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN)
+				).then(
+						getTxnRecord(transferTxn).hasPriority(recordWith()
+								.autoAssociated(accountTokenPairs(List.of()))),
+						getAccountInfo(beneficiary)
+								.hasAlreadyUsedAutomaticAssociations(0)
+								.has(accountWith().noChangesFromSnapshot(beneficiary)),
+						getAccountInfo(treasury)
+								.hasAlreadyUsedAutomaticAssociations(0)
+								.has(accountWith().noChangesFromSnapshot(treasury)),
+						/* The treasury should still have an open auto-association slots */
+						cryptoTransfer(
+								moving(500, otherFungibleToken).between(beneficiary, treasury)
+						)
+				);
+	}
+
+	public HapiApiSpec failedAutoAssociationHasNoSideEffectsOrHistoryForUnrelatedProblem() {
+		final var treasury = "treasury";
+		final var beneficiary = "beneficiary";
+		final var unluckyBeneficiary = "unluckyBeneficiary";
+		final var uniqueToken = "unique";
+		final var fungibleToken = "fungible";
+		final var multiPurpose = "multiPurpose";
+		final var transferTxn = "transferTxn";
+
+		return defaultHapiSpec("failedAutoAssociationHasNoSideEffectsOrHistory")
+				.given(
+						newKeyNamed(multiPurpose),
+						cryptoCreate(treasury),
+						tokenCreate(fungibleToken)
+								.tokenType(TokenType.FUNGIBLE_COMMON)
+								.initialSupply(1_000L)
+								.treasury(treasury),
+						tokenCreate(uniqueToken)
+								.tokenType(NON_FUNGIBLE_UNIQUE)
+								.supplyKey(multiPurpose)
+								.initialSupply(0L)
+								.treasury(treasury),
+						mintToken(uniqueToken, List.of(copyFromUtf8("ONE"), copyFromUtf8("TWO"))),
+						cryptoCreate(beneficiary).maxAutomaticTokenAssociations(2),
+						cryptoCreate(unluckyBeneficiary),
+						tokenAssociate(unluckyBeneficiary, uniqueToken),
+						getAccountInfo(beneficiary).savingSnapshot(beneficiary),
+						getAccountInfo(unluckyBeneficiary).savingSnapshot(unluckyBeneficiary)
+				).when(
+						cryptoTransfer(
+								movingUnique(uniqueToken, 1L)
+										.between(treasury, beneficiary),
+								moving(500, fungibleToken).between(treasury, beneficiary),
+								movingUnique(uniqueToken, 1L)
+										.between(treasury, unluckyBeneficiary)
+						).via(transferTxn).hasKnownStatus(SENDER_DOES_NOT_OWN_NFT_SERIAL_NO)
```

Comment:
nit: `SENDER_DOES_NOT_OWN_NFT_SERIAL_NO` this could be `SENDER_DOES_NOT_OWN_NFT_SERIAL_NUM`. This can be modified in next PR for protobufs.

Reply:
Agreed. 😁